src/validation.cpp

```diff
@@ -739,7 +739,15 @@ bool MemPoolAccept::AcceptSingleTransaction(const CTransactionRef &ptx,
         return false;
     }
 
-    GetMainSignals().TransactionAddedToMempool(ptx);
+    std::vector<Coin> spent_coins;
+    spent_coins.reserve(ptx->vin.size());
```

Comment:
This seems to be done a lot, might be worth creating a function on the transaction classes.